### PR TITLE
Updated row_number and ntile to include a 'ties' argument that is pas…

### PR DESCRIPTION
**Pull request by Inferrator**
…sed to base::rank' 'ties.method' argument.

It still defaults to 'first', but there are instances where this is not appropriate. One example would be RFM analysis, where 'min' or 'max' are more suitable options.

### CODE REVIEW

**Comment by Inferrator** [comment]
AppVeyor:

`Error in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) : 
  namespace 'rlang' 0.3.4 is being loaded, but >= 0.4.0 is required`

Has nothing to do with the PR.

**Comment by Inferrator** [comment]
Is anyone still looking at PRs sometime?

File: R/rank.R [hunk 1]
```diff
@@ -26,6 +26,9 @@
 #' @param x a vector of values to rank. Missing values are left as is.
 #'   If you want to treat them as the smallest or largest values, replace
 #'   with Inf or -Inf before ranking.
+#' @param ties a character string specifying how ties are treated, one of
+#' "average", "first", "last", "random", "max", "min". For details, see
+#' base::rank.
```
Comment (romainfrancois) [inline R/rank.R]:
```suggestion
#' [base::rank()].
```

File: R/rank.R [hunk 2]
```diff
@@ -45,27 +48,30 @@ NULL
...
   }
 
 }
 
 # Definition from
 # http://blogs.msdn.com/b/craigfr/archive/2008/03/31/ranking-functions-rank-dense-rank-and-ntile.aspx
 #' @param n number of groups to split up into.
+#' @param ties a character string specifying how ties are treated, one of
+#' "average", "first", "last", "random", "max", "min". For details, see
+#' base::rank.
```
Comment (romainfrancois) [inline R/rank.R]:
```suggestion
#' [base::rank()].
```

**Comment by Inferrator** [comment]
Thanks Romain!
I'd love to run devtools::documents(), but it doesn't work. Maybe some can tell Hadley to finally fix Roxygen2. He wouldn't even listen to Dirk (see: https://github.com/r-lib/roxygen2/issues/822).

**Comment by Inferrator** [comment]
oops, sorry.

**Comment by hadley** [comment]
Sorry I didn't see this earlier — these functions are deliberately designed to be narrower than `base::rank()`, so I don't think we should be passing these arguments on. You can always call `rank()` directly if you want them :smile:

**Comment by Inferrator** [comment]
hadley: WTH! This is bad. You don't provide a reason. You create functions that are deliberately bad and inconsistent with base R. You refuse meaningful community input. You are ruining R.

**Pull request closed without merging**